statement.go

```diff
@@ -470,6 +470,82 @@ func (s *Stmt) ExecContext(ctx context.Context, nargs []driver.NamedValue) (driv
 	return &result{ra}, nil
 }
 
+// ColumnCount returns the number of columns that will be returned by executing the prepared statement.
+// If any of the column types is invalid (which can happen when the type is ambiguous), the result will be 1.
+// Returns an error if the statement is closed or uninitialized.
+func (s *Stmt) ColumnCount() (int, error) {
+	if s.closed {
+		return 0, errClosedStmt
+	}
+	if s.preparedStmt == nil {
+		return 0, errUninitializedStmt
+	}
+
+	count := mapping.PreparedStatementColumnCount(*s.preparedStmt)
+	return int(count), nil
+}
+
+// ColumnTypes returns the types of all columns in the result set of the prepared statement.
+// Returns an error if the statement is closed or uninitialized.
+func (s *Stmt) ColumnTypes() ([]Type, error) {
```

Comment:
Can we also add tests here for the error paths and the happy path (once rewritten to `GetType`)?